src/Stratis.Bitcoin/P2P/ISelfEndpointTracker.cs

```diff
@@ -7,6 +11,15 @@ namespace Stratis.Bitcoin.P2P
     /// </summary>
     public interface ISelfEndpointTracker
     {
+        /// <summary>Update external IP address and peer score of the node.</summary>
+        void UpdateAndAssignMyExternalAddress(IPEndPoint ipEndPoint, int? ipEndPointPeerScore);
+
+        /// <summary>External IP address of the node.</summary>
+        IPEndPoint MyExternalAddress { get; set; }
+
+        /// <summary>Whether external IP address of the node is final or can be updated.</summary>
+        bool IsMyExternalAddressFinal { get; set; }
```

Comment:
This guy should not be an interface if we update everything through update method and the guy above should only have public getter